scripts/electron/preload.js

```diff
@@ -15,22 +17,81 @@ let certHandler;
 let cookies = '';
 
 
+/**
+ * If this is the main window.
+ * @type {boolean}
+ */
+let isMain = false;
+
+
+/**
+ * The copied base settings file that will be loaded by the application.
+ * @type {string}
+ */
+let baseSettingsFile = '';
+
+
+/**
+ * Settings files available to the application.
+ * @type {!Array<!ElectronOS.SettingsFile>}
+ */
+let settingsFiles = [];
+
+
+/**
+ * The directory containing user config files and copied app settings.
+ * @type {string}
+ */
+let userSettingsDir = '';
+
+
+/**
+ * If user settings are supported.
+ * @type {boolean}
+ */
+let userSettingsSupported = false;
+
+
 /**
  * General event types.
  * @enum {string}
  */
 const EventType = {
+  IS_MAIN: 'is-main-window',
   UPDATE_CHECK: 'check-for-updates',
 
   CERT_HANDLER_REGISTERED: 'client-certificate-handler-registered',
   CERT_SELECT: 'select-client-certificate',
   CERT_SELECTED: 'client-certificate-selected',
 
   COOKIE_SET: 'set-cookie',
-  COOKIE_UPDATE: 'update-cookies'
+  COOKIE_UPDATE: 'update-cookies',
+
+  SETTINGS_ADD: 'add-settings',
+  SETTINGS_GET_FILES: 'get-settings-files',
+  SETTINGS_GET_BASE_FILE: 'get-base-settings-file',
+  SETTINGS_GET_USER_DIR: 'get-user-settings-dir',
+  SETTINGS_REMOVE: 'remove-settings',
+  SETTINGS_SET: 'set-settings',
+  SETTINGS_SUPPORTED: 'user-settings-supported',
+  SETTINGS_UPDATE: 'update-settings'
```

Comment:
Are these event types something that can be exported from the Electron context/otherwise shared rather than redeclaring identical strings?

Reply:
Yeah that was bothering me and I never fixed it. Moved the cookie/settings event types that had duplicated enums to their own modules, that are now used by the app and the preload script.